### PR TITLE
路線判定: 並走・分岐区間のエッジケース回帰テストを追加

### PR DESCRIPTION
**Pull request by TakuroFukamizu**
## 背景

並走から分離・合流する区間や、新幹線と在来線が並ぶ区間では路線判定が揺れやすい。`MapMatcher` を今後改善する際にデグレを検出できるよう、実データの線形で再現する回帰テストを用意した。

## 変更

- `src/scripts/build-route-edge-fixture.ts`（`pnpm fixture:route-edge`）: 公開データセット v1.4.0 の H3 タイルから、対象区間の周囲 1.6km の全路線（地下鉄含む）を `tests/railway/route-edge-cases/fixtures/` に切り出す。出典（国土数値情報 N02-23）を記載。
  - v1.4.0 には東海道新幹線 東京〜品川が存在しないため、東海道線の線形を東へ 35m ずらした合成セグメントを追加している。
- `tests/railway/route-edge-cases/`
  - `fixture-db.ts`: 本番の `DexieRailwayDatabase.findSegmentsNear` と同じ抽出条件のインメモリDB
  - `trace.ts`: シード固定の 1Hz 合成GPS（台形速度、駅停車は駅座標、横方向ノイズ σ6m、データ欠落の直線補間は 1.5km まで）
  - `scenarios.ts`: 10 シナリオ ＋ 自動生成の逆方向（田端・品川の分岐、東京〜品川、東京〜上野、御茶ノ水→両国→亀戸、総武快速の地下区間）
  - `runner.ts`: `DEFAULT_TRACKING_CONFIG` で実行し、表示路線を毎秒評価
  - `route-edge-cases.test.ts`: ハードチェック（禁止路線なし／最終路線が正しい／誤表示なし）と挙動のスナップショット。現状で通らないチェックは `KNOWN_ISSUES` として `it.fails` で実行し、改善されると赤くなる
  - `README.md`: 使い方と結果の読み方
- `package.json`: `test:route-edge`, `fixture:route-edge`
- `.claude/agents/route-edge-regression.md`: このテストの実行・結果評価・テスト改善を行う専用サブエージェント

## テストで判明した既知の誤判定（ロジックは未変更）

- 田端: 西日暮里〜田端から「東北新幹線（大宮〜東京）」に切り替わり、京浜東北線北行きと合流側では戻らない
- 東京〜上野: 同梱サンプルの直線近似「JR東北新幹線」を表示し続ける
- 品川付近で京急「本線」、東京〜新橋で「有楽町線」、御茶ノ水で丸ノ内線、両国で総武快速線を一時的に表示

データ面では、東海道新幹線 東京〜品川の欠落と、東北新幹線 東京〜上野の二重登録（同梱サンプル＋MLIT）も確認した。修正は別 Issue / PR で行う。

## テスト

```
pnpm test:route-edge   Tests 83 passed
pnpm test              Test Files 41 passed / Tests 399 passed
pnpm lint / tsc --noEmit: エラーなし
Node 24 / 26（CI=true）で同一結果
```

設定値を 1 つ変える（`routeChallengerMinMargin` 12→6）と 13 件が失敗することを確認済み。

## レビューで出た改善点（別途対応予定）

Codex（gpt-6-astra）レビューで、駅座標への延長による折り返し、本番HUDの路線喪失猶予の未反映、連続誤表示時間の未記録、トンネル区間での測位欠落の未検証などが指摘されている。

🤖 Generated with [Claude Code](https://claude.com/claude-code)

https://claude.ai/code/session_0141t9HP7dCpCJ1Mx1qiwz3b

### CODE REVIEW

File: src/scripts/build-route-edge-fixture.ts [hunk 1]
```diff
@@ -0,0 +1,297 @@
...
+  const outputPath = fileURLToPath(OUTPUT_URL);
+  fs.mkdirSync(path.dirname(outputPath), { recursive: true });
+  fs.writeFileSync(outputPath, `${JSON.stringify(fixture)}\n`);
+  console.log(
+    `[route-edge-fixture] v${version}: ${fixture.lines.length} lines, ${fixture.stations.length} stations, ` +
+      `${fixture.segments.length} segments -> ${path.relative(process.cwd(), outputPath)}`
+  );
+}
+
+if (process.argv[1]?.includes('build-route-edge-fixture')) {
```
[the remark this answers is not in the document]
Comment (TakuroFukamizu) [reply src/scripts/build-route-edge-fixture.ts]: 指摘のとおり、他のスクリプトと判定方法が揃っていませんでした。`import.meta.url.endsWith(process.argv[1])` を併用する形に修正しました（0ef3539）。

File: tests/railway/route-edge-cases/trace.ts [hunk 2]
```diff
@@ -0,0 +1,248 @@
...
+  runs: TraceRun[],
+  options: { initialDwellS?: number; startTimestampMs?: number } = {}
+): TracePoint[] {
+  const random = createRandom(seed);
+  const points: TracePoint[] = [];
+  let timestampMs = options.startTimestampMs ?? 1_000;
+  let previousEnd: LatLon | null = null;
+
+  const emitStationary = (at: LatLon, seconds: number, runIndex: number, phase: TracePoint['phase'], gps: GpsQuality, secondsIntoRun: number) => {
+    for (let i = 0; i < seconds; i++) {
```
[the remark this answers is not in the document]
Comment (TakuroFukamizu) [reply tests/railway/route-edge-cases/trace.ts]: 修正しました（0ef3539）。1Hz のトレースなので `Math.max(0, Math.round(seconds))` でチック数を整数に丸めています。

File: tests/railway/route-edge-cases/trace.ts [hunk 3]
```diff
@@ -0,0 +1,248 @@
...
+    const path = buildRunPolyline(db, run.path, previousEnd, runs[runIndex + 1]?.path[0] ?? null);
+    // Stops happen at platforms, not at polyline ends (which can sit hundreds of metres away on a junction).
+    if (runIndex === 0) {
+      const station = nearestSegmentStation(db, run.path[0], path[0]);
+      if (station) path.unshift(station);
+    }
+    if (run.dwellAtEndS) {
+      const station = nearestSegmentStation(db, run.path[run.path.length - 1], path[path.length - 1]);
+      if (station) path.push(station);
+    }
```
[the remark this answers is not in the document]
Comment (TakuroFukamizu) [reply tests/railway/route-edge-cases/trace.ts]: 修正しました（0ef3539）。駅スナップと停車サンプル生成の両方を `!== undefined` 判定に統一し、`0` は「長さ0の停車」として `endsAtRest` の扱いと一致するようにしました。

File: tests/railway/route-edge-cases/runner.ts [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+import { RouteLockState, shouldDisplaySelectedRoute } from '../../../src/domain/models/railway';
+import { MapMatcher } from '../../../src/domain/railway/map-matcher';
+import { FixtureRailwayDb } from './fixture-db';
+import { EdgeScenario } from './scenarios';
+import { generateTrace } from './trace';
+
+export type Tick = {
+  t: number;
+  runIndex: number;
+  phase: string;
```
[the remark this answers is not in the document]
Comment (TakuroFukamizu) [reply tests/railway/route-edge-cases/runner.ts]: 修正しました（0ef3539）。`Tick.phase` を `TracePoint['phase']` にして union を維持しています。
